### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by neoascetic**
When I did #43 and dig into webpage's code, I felt pain.
Let's make Terasology splash site as beatiful as Terasology itself!
You can view demo of these changes [here](http://neoascetic.me/terasology.org/), so please report on any bugs you can find.

I've optimized all pictures (to speed up their loading) using ImageOptim and because of this the PR is so fat in terms of bytes (+3Mb). If you do not want this, I can remove these changes and make a force-push to the branch again to prevent repo from bloating up.

### CODE REVIEW

File: scripts/main.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    $(window).on('hashchange load', function() {
+        $menuEntries.hide();
+        var clickedMenuEntry =
+            $menuEntries.filter(window.location.hash).size() ?
+            window.location.hash :
+            '#index';
+        $(clickedMenuEntry).show();
+    });
+
+    var curBackground = 1;
```
Comment (small-jeeper) [inline scripts/main.js:33]: I think it would be better, if you move all 'var' to the beginning of the function =)

**Comment by neoascetic** [comment]
I and Steve McConnell don't think so. Moving all variables definitions to the beginning will increase their "livetime" and, thus, will decrease readability. So, let them stay where they are :)

**Comment by neoascetic** [comment]
Bug. Need to trigger `hashchange` event right within `document.ready` handler. Waiting to `window.load` takes too much time. Also, preloading of background images and their rotating must be moved to `window.load` handler. This will increase page loading speed and, therefore, UX.

**Comment by small-jeeper** [comment]
> Moving all variables definitions to the beginning will increase their "livetime", and, thus, will decrease readability.

if I define all the variables in the top of function, why will it increase their "livetime"? O.o Can you explain me? =)

**Comment by Cervator** [comment]
I'm thinking it is from the Code Complete book and relates to local variables that are declared further from where they're needed than they have to be. So it would be slightly more effort to scroll back up and double check or change the initial value, for instance :-)

@neoascetic - nice work! This looks like a great pile of changes, especially finally getting better buttons :D

Works great in Firefox and Chrome, IE11 for me isn't loading the buttons although now finally the background cycles. IE9 seems to load buttons OK. 

Loaded _immediately_ via phone browser, very snappy!

One other problem - it seems the new architecture tries to preload part of the applet somehow, resulting in a security popup. Applet doesn't actually launch until you go to the "Play" page though.

I'm not sure if that alert will go away when we're finally signing the applet properly, if it does then maybe it is okay. Or is there any easy way to suppress that until you actually click the play button? That's the main obstacle to accepting this PR right now :-)

And yeah looks like we're missing grayscale version of the Jenkins and blog buttons (I figure might as well include them, even on a splash site). Looks good to have them toggle between gray and color! Old version did a sort of scaled highlight where the button grows slightly on-hover, is that possible too or an ajax magic thing?

Right click also works now, yay :-)

Asked on IRC and @skaldarnar noted missing non-hover social buttons for Twitter, G+, YouTube, and Reddit (appear fine on hover). He's on Linux using FF 26. Would be interesting to see if anybody else gets that too. Chromium worked fine.

So questions/issues in summary:
- [x] Applet causes an alert on loading the main page, really shouldn't until the "Play" button is clicked (important)
- [x] IE11 doesn't load menu buttons, IE9 does, other browsers seem fine
- [ ] Need gray versions of Jenkins + blog image (minor / pre-existing, can live without)
- [ ] On-hover scaling trick for social buttons? (minor and might be worth the trade-off)
- [ ] Maybe an issue with some of the non-hover social buttons

**Comment by Cervator** [comment]
Oh and the little site icon isn't showing up for me, but then might be from the demo site setup :-)

I don't get the applet warning in my Firefox but suspect that's because I've previously "remembered" to accept anything for our applet.

Looking forward to accepting this! Many thanks :-)

Edit: Odd, the buttons in my laptop IE 9 eventually disappeared and now I can't force-refresh them back.

Edit2: Forgot to mention #18 - clicking the "Terasology" logo used to hide everything leaving just the backgrounds visible and cycling. Which was sort of a nice touch, but confusing on the logo. #18 could be used to add a single little "make invisible" button in a lower corner, not really a needed fix for this PR :-)

**Comment by neoascetic** [comment]
Yeah, variable's "livetime" is from Code Complete :)

Is there need to "Jenkins" button at all? Since it is completely technical stuff. 
Also, blog wasn't updated a half of a year, do we still need it?

On-hover buttons scaling trick. Maybe try to make Splash Site's buttons looks like in a launcher (yellow on hover, without scalling)? I more prefer this idea.

Should I also revert back logo since it is not the same as in the launcher?

Viewing background images in "fullscreen" without menu buttons and stuff is cool, but clicking on title for this is too unintuitive. I also don't like any kind of button for this. I think that clicking on background image itself for switching to the "fullscreen" mode is simple and pretty intuitive solution. How about that?

**Comment by neoascetic** [comment]
@Cervator please see the updated demo version!

**Comment by neoascetic** [comment]
Need to move scripts to the bottom of the page, this probably will fix the issue with buttons backgrounds.

**Comment by Cervator** [comment]
Great! Able to confirm applet warning gone and buttons working in IE, that's enough to merge this!

Don't forget to add yourself to the credits / contributors though! And feel free to do that scripts thing if that helps in a new PR :-)

On Jenkins and blog ... yeah they aren't really needed. I just like having more icons, haha

Had another user check on Linux and all was well there. Background fade on click anywhere is perfect!

I'll go through the issues after merging this since you probably fixed half of them (or more) in one go :-)

**Pull request merged**